src/Chapter3.hs

```diff
@@ -560,22 +638,46 @@ introducing extra newtypes.
 🕯 HINT: if you complete this task properly, you don't need to change the
     implementation of the "hitPlayer" function at all!
 -}
+newtype Health = Health
+    { getHealth :: Int
+    }
+newtype Armor = Armor
+    { getArmor :: Int
+    }
+newtype Attack = Attack
+    { getAttack :: Int
+    }
+newtype Dexterity = Dexterity
+    { getDexterity :: Int
+    }
+newtype Strength = Strength
+    { getStrength :: Int
+    }
+newtype Damage = Damage
+    { getDamage :: Int
+    }
+newtype Defense = Defense
+    { getDefense :: Int
+    }
+
 data Player = Player
-    { playerHealth    :: Int
-    , playerArmor     :: Int
-    , playerAttack    :: Int
-    , playerDexterity :: Int
-    , playerStrength  :: Int
+    { playerHealth    :: Health
+    , playerArmor     :: Armor
+    , playerAttack    :: Attack
+    , playerDexterity :: Dexterity
+    , playerStrength  :: Strength
     }
 
-calculatePlayerDamage :: Int -> Int -> Int
-calculatePlayerDamage attack strength = attack + strength
+calculatePlayerDamage :: Attack -> Strength -> Damage
+calculatePlayerDamage attack strength = Damage (getAttack attack + getStrength strength)
 
-calculatePlayerDefense :: Int -> Int -> Int
-calculatePlayerDefense armor dexterity = armor * dexterity
+calculatePlayerDefense :: Armor -> Dexterity -> Defense
+calculatePlayerDefense armor dexterity =
+    Defense (getArmor armor * getDexterity dexterity)
 
-calculatePlayerHit :: Int -> Int -> Int -> Int
-calculatePlayerHit damage defense health = health + defense - damage
+calculatePlayerHit :: Damage -> Defense -> Health -> Health
+calculatePlayerHit damage defense health =
+    Health (getHealth health + getDefense defense - getDamage damage)
```

Comment:
Nice work on newtypes! :medal_sports: Newtypes are one very important feature in Haskell :+1:
You can read more about the newtype pattern in our blog post: https://kowainik.github.io/posts/haskell-mini-patterns#newtype